### ISSUE TITLE
\hideinheritancegraph not working when using out of place documentation

### ISSUE THREAD

**Issue by TheNicker** [doxygen/doxygen#10460]
**Describe the bug**
`\hideinheritancegraph` seems to not-hide the inheritance graph both in CPP and Python

**Expected behavior**
Adding `\hideinheritancegraph` to the comment block should output html without inheritance graph

**To Reproduce**
Install graphviz, add `HAVE_DOT = YES` to the default doxygen configuration file and run doxygen on the following code
Supplied a self contained example that reproduces the issue: [doxytest.zip](https://github.com/doxygen/doxygen/files/13513807/doxytest.zip)

```
#include <string>

/** 
    \class ClassBase
    \hideinheritancegraph
    \brief This is the base class that does nothing
*/

class ClassBase
{
    std::string getName() { return "gles";}
};
        
class ClassDerivedA : public ClassBase{
};


class ClassDerivedB : public ClassBase{
};
```

**Version**
1.9.3 from ubuntu 22.03 official repository (jammy) and 1.10 locally compiled (latest commit).
Platform: Ubuntu 22.04.3 LTS
Architecture: 64 bit

**Comment by albert-github** [comment]
That it is not working for doxygen version 1.9.3 is logical as it is only recently added to the doxygen master. With the 1.9.3 version you should also get a warning like:
```
.../test.cpp:5: warning: Found unknown command '\hideinheritancegraph'
```
The fact that it does not work with the current master version is due to the out of place documentation of `ClassBase` (i.e. the statement `class ClassBase`) which is in this case not necessary as the documentation is straight in front of the class.
It also works when you leave the out of pace documentation as it is bu add also 
```
/** \hideinheritancegraph */ just before the class.
```

That said it should also be working for the plain out of place documentation.

**Comment by TheNicker** [comment]
It doesn't work for `in place` or `out of place` documentation, to verify again just tried it with 1.10.0 (c87ad3cbfbfe67a14e5078250178478719815dbb) with `in-place` documentation

**Comment by albert-github** [comment]
With the following I was successful:
```
#include <string>


/**
    \class ClassBase
    \brief This is the base class That does nothing
*/

///    \hideinheritancegraph
class ClassBase
{
    std::string getName() { return "gles";}
};

class ClassDerivedA : public ClassBase
{

};

class ClassDerivedB : public ClassBase
{

};

class ClassDerivedC : public ClassBase
{

};
```
and the settings:
```
QUIET = YES
HAVE_DOT = YES
```

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10461

**Comment by TheNicker** [comment]
The last two works for me as well  with 1.10, I guess I didn't test `in place` with 1.10.
So I guess it's just an issue with the `out of place` documentation

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
